opentelemetry/src/context.rs

```diff
@@ -328,19 +330,183 @@ impl Context {
+    ///
+    /// let unsuppressed_cx = cx.with_telemetry_unsuppressed();
+    /// assert_eq!(unsuppressed_cx.is_telemetry_suppressed(), false);
+    /// ```
+    pub fn with_telemetry_unsuppressed(&self) -> Self {
+        Context {
+            entries: self.entries.clone(),
+            #[cfg(feature = "trace")]
+            span: self.span.clone(),
+            suppress_telemetry: false,
+        }
+    }
+
+    /// Returns a clone of the current thread's context with telemetry suppression enabled.
+    ///
+    /// This is a more efficient form of `Context::current().with_telemetry_suppressed()`
+    /// as it avoids the intermediate context clone.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use opentelemetry::Context;
+    ///
+    /// // Get a suppressed context based on the current one
+    /// let suppressed = Context::current_with_telemetry_suppressed();
+    /// assert_eq!(suppressed.is_telemetry_suppressed(), true);
+    /// ```
+    pub fn current_with_telemetry_suppressed() -> Self {
+        Self::map_current(|cx| cx.with_telemetry_suppressed())
+    }
+
+    /// Returns a clone of the current thread's context with telemetry suppression disabled.
+    ///
+    /// This is a more efficient form of `Context::current().with_telemetry_unsuppressed()`
+    /// as it avoids the intermediate context clone.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use opentelemetry::Context;
+    ///
+    /// // Get an unsuppressed context based on the current one
+    /// let unsuppressed = Context::current_with_telemetry_unsuppressed();
+    /// assert_eq!(unsuppressed.is_telemetry_suppressed(), false);
+    /// ```
+    pub fn current_with_telemetry_unsuppressed() -> Self {
+        Self::map_current(|cx| cx.with_telemetry_unsuppressed())
+    }
+
+    /// Enters a scope where telemetry is suppressed.
+    ///
+    /// This is a convenience method that creates a new context with telemetry
+    /// suppression enabled and attaches it.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use opentelemetry::Context;
+    ///
+    /// let _guard = Context::enter_suppressed();
+    /// assert_eq!(Context::current().is_telemetry_suppressed(), true);
+    /// ```
+    pub fn enter_suppressed() -> ContextGuard {
+        Self::current_with_telemetry_suppressed().attach()
+    }
+
+    /// Returns whether telemetry is suppressed in the current context.
+    ///
+    /// This is a convenience method that combines `Context::current()` and
+    /// `is_telemetry_suppressed()`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use opentelemetry::Context;
+    ///
+    /// // Default context has telemetry suppression disabled
+    /// assert_eq!(Context::is_current_suppressed(), false);
+    ///
+    /// // Enter a suppressed scope
+    /// let _guard = Context::enter_suppressed();
+    /// assert_eq!(Context::is_current_suppressed(), true);
+    /// ```
+    #[inline]
+    pub fn is_current_suppressed() -> bool {
```

Comment:
maybe I can start with just this method and `enter_unsuppressed` only in public API, and keep others non-public... That should do the job for now.

Reply:
for the initial need, this can be in the SDK, not in the API.
There is a possible use case for this to be in the API, which is instrumentations of higher layer want to suppress underlying layer. (like reqwest wanting to suppress hyper..)..
But I am okay with start with SDK to keep the API lighter. And move to API when the actual need comes.

Reply:
From community call today: will keep in API itself, to avoid any future movement.